app/controllers/optOut/ConfirmOptOutController.scala

```diff
@@ -45,40 +47,41 @@ class ConfirmOptOutController @Inject()(view: ConfirmOptOut,
 
   private val errorHandler = (isAgent: Boolean) => if (isAgent) itvcErrorHandlerAgent else itvcErrorHandler
 
-  private def withOptOutQualifiedTaxYear(isAgent: Boolean)(function: OptOutCheckpointViewModel => Result)
-                                        (implicit mtdItUser: MtdItUser[_]): Future[Result] = {
-
-    optOutService.optOutCheckPointPageViewModel().map {
-      case Some(optOutOneYearCheckpointViewModel) => function(optOutOneYearCheckpointViewModel)
-      case None =>
-        Logger("application").error("No qualified tax year available for opt out")
-        errorHandler(isAgent).showInternalServerError()
-    }
-
-  }
-
-  private def withRecover(isAgent: Boolean)(code: => Future[Result])(implicit mtdItUser: MtdItUser[_]): Future[Result] = {
-    code.recover {
-      case ex: Exception =>
-        Logger("application").error(s"request failed :: $ex")
-        errorHandler(isAgent).showInternalServerError()
-    }
-  }
-
   def show(isAgent: Boolean): Action[AnyContent] = auth.authenticatedAction(isAgent) {
     implicit user =>
       withRecover(isAgent) {
-        withOptOutQualifiedTaxYear(isAgent)(
-          viewModel => Ok(view(viewModel, isAgent = isAgent))
-        )
+
+        val resultToReturn = for {
+          viewModel <- OptionT(optOutService.optOutCheckPointPageViewModel())
+          result <- OptionT(Future.successful(Option(toPropositionView(isAgent, viewModel))))
```

Comment:
should fix: use `Some`  eg. `Option(toPropositionView(isAgent, viewModel))` -> `Some(toPropositionView(isAgent, viewModel))` to represent value

Reply:
won't compile will show you, I know a bit strange

Reply:
done, have to use Option due to OptionT being Invariance class type